spec/controllers/admin/orders_controller_spec.rb

```diff
@@ -0,0 +1,206 @@
+require "rails_helper"
+include SessionsHelper
+
+RSpec.describe Admin::OrdersController, type: :controller do
+  describe "GET #index" do
+    context "when not logged in" do
+      before do
+        get :index
+      end
+
+      it "redirect to root_url" do
+        expect(response).to redirect_to login_url
+      end
+    end
+
+    context "when user not permission" do
+      let!(:user) {FactoryBot.create :user, role: false}
+      before do
+        log_in user
+        get :index
+      end
+
+      it "display flash danger" do
+        expect(flash[:danger]).to eq I18n.t("not_permission")
+      end
+      it "redirect to root_url" do
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe "When logged in and has permission" do
+    let(:user) {FactoryBot.create :user, role: true}
+    before do
+      log_in user
+    end
+
+    describe "GET #index" do
```

Comment:
Cái line code này nên ở ngoài cái chỗ check login not permission với has permission nhé